.github/workflows/docker-ci.yaml

```diff
@@ -0,0 +1,39 @@
+# Example pulled from https://docs.docker.com/guides/gha/#conclusion
+
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Extract Docker image metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ vars.DOCKER_USERNAME }}/my-image
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ vars.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v6
+        with:
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
Won't this always be true? What's the benefit of having it here?

Should we instead prevent running this on pull requests?

Reply:
a) It's a best practices example that I copy-pasta'd.

b) It is designed to test build any potential new releases before merge to verify that it passes the build process. This flag prevents publishing those test builds to Docker Hub.

c) I have already pulled out the hook that runs this build process on pull request. This CI process now only runs on merge with main. So yes, in effect this now looks for a condition that is always true. It may be unnecessary now but it also doesn't hurt and it prevents making a mess should the hooks change.